### ISSUE TITLE
How to use ? 

### ISSUE THREAD

**Issue by minanagehsalalma** [4zv4l/GoDeauth#1]

**Comment by 4zv4l** [comment]
It's not finished yet 
You can look at the code and propose update or modification but for now except showing the interfaces and scanning for lan Mac addresses it doesn't really do anything

**Comment by 4zv4l** [comment]
Usually type "h" to get some help in the program

**Comment by minanagehsalalma** [comment]
> It's not finished yet You can look at the code and propose update or modification but for now except showing the interfaces and scanning for lan Mac addresses it doesn't really do anything

Hmmm why lan Macs ? 
Isn't this meant for WIFI ? 

I also don't know GO language so ... 

How are you planning to implement it ? what lab is available for Go to do packets and such like scapy for python ?

**Comment by 4zv4l** [comment]
Lan macs because I didn't find yet how to set the interface in monitor mode and to scan for bssid
I didn't work on that project due to my exam period but I'll continue when I'm done
I still need to learn how deauth packages work
And how to do that in Go

**Comment by minanagehsalalma** [comment]
> Lan macs because I didn't find yet how to set the interface in monitor mode and to scan for bssid

@4zv4l  Hmm ... Then why you even thought about doing this in go ? 

What was the motive ?

**Comment by 4zv4l** [comment]
Well that's my repository I do what I want 😂

**Comment by minanagehsalalma** [comment]
> Well that's my repository I do what I want 😂

@4zv4l  Lol , I was just asking man 
jess 

As i am interested too 
so... peace

**Comment by 4zv4l** [comment]
I do it in Go because I like Go.
Bettercap is made in Go so i thought this could be possible, I'm not working on it for now, I didn't think about all yet if you wanna know.
So if you wanna help well you can open a pull request or give some tips.
I know this would be way easier in python, I saw lot of example but that's not what I want, I want something easy to use, easy to compile on many platforms and in Go (because once again I really like Go xD)

**Comment by minanagehsalalma** [comment]
> Bettercap is made in Go so i thought this could be possible

Oh that's a good point 

You can see how they deauth from there but i doubt that they do it without external software or something idk 

> I want something easy to use, easy to compile on many platforms and in Go (because once again I really like Go xD)

Even if it were to compile to diff platforms  i doubt that it would work on there 
As monitor mode is pretty complicated 

But seems like a good idea

I am following the repo

**Comment by 4zv4l** [comment]
Thanks !
Yeah I looked and try to do something similar but I would like to make something really "independent" without having to set the interface in monitor mode before or having to get the bssid before, because I could choose to just have a prompt for a Mac addresse and asking to set the interface in monitor mode and voila send the deauth but I will try to make something that does "all the job"
I mean if aircrack can set an interface in monitor mode I guess a Go program can do it too

**Comment by minanagehsalalma** [comment]
> I mean if aircrack can set an interface in monitor mode I guess a Go program can do it too

Hmm But why Tho ? 

Aircrack does it just fine 
This just complicating for no reason 
You can even do it using the iw command in Linux 


You know what hasn't been done before ? 

Deauth on windows 
It's already available using a tool like common view for Wi-Fi but its implementation sucks  

it can be done using [Microsoft network monitor tool](https://docs.microsoft.com/en-us/windows-hardware/drivers/portable/using-the-netmon-tool) But it needs automation to be practical

**Comment by 4zv4l** [comment]
Yeah I could do it using system command but on Linux what if iw isn't installed ? My daily driver OS is Alpine Linux and I don't use iw so I will first try without using system command
And if I succeed without using system command then it should also work o Windows with cross compilation
Also if it doesn't use system command I can port it using tinyGo on a microcontroller having a wireless card

**Comment by minanagehsalalma** [comment]
> if I succeed without using system command then it should also work o Windows with cross compilation

@4zv4l hmm are you sure about that?  Windows is built different so

Monitor mode is way complicated there tho

**Comment by TotallyNotAHaxxer** [comment]
Good evening/night @4zv4l my name is ArkAngeL :D "  i have read alot in this and i can say im quite interested in the project, go is a great choice with this however correct me if i am wrong but from my experience you can go about finding the interface names with gopacket or something, ive made some wireless network frameworks in the past with go and instead of using IW, and worrying about the operating system you can just run inline C with go, or have gopacket do the work for you in the following code example 

```go
package main

import (
    "fmt"
    "log"
    "github.com/google/gopacket/pcap"
)

func main() {
    // Find all devices
    devices, err := pcap.FindAllDevs()
    if err != nil {
        log.Fatal(err)
    }

    // Print device information
    fmt.Println("Devices found:")
    for _, device := range devices {
        fmt.Println("\nName: ", device.Name)
        fmt.Println("Description: ", device.Description)
        fmt.Println("Devices addresses: ", device.Description)
        for _, address := range device.Addresses {
            fmt.Println("- IP address: ", address.IP)
            fmt.Println("- Subnet mask: ", address.Netmask)
        }
    }
}
```

it could as well be done alot easier, i figure there can be alot more such as root user requirements or regex to find the correct interface the person can use with this script, i feel it is too large to just commit and make pull requests to since well its in general quite alot to just up and add, but i would for sure say experiment with it, and if you wanted to find a way to verify the mac addresses, capture SSID's by sending out probe packets ( If i am getting that correct, excuse me if i am wrong on that terminology i have not been messing with wifi recently ) to grab information on the wifi names, which kinda turn it into a whole new module, i like the idea of choosing go as well given go is a very fast language and is one of the eaiser languages to write in if you factor in  C, C++, LASM ( Linux Assembly ), fortran, etc etc the list goes on :joy: 


you said at the top

```
I still need to learn how deauth packages work
And how to do that in Go
```

im not sure if there is one standard that the go devs made for wifi, but from what i see you are already using gopacket and i would think that would be your best bet here given you can customize streams, buffers, BPF ( IF you are doing stuff like captring EOPAL or whatever ), listeners, etc etc.

# the c example for getting interface names 

> go has an awsome way of inline C, and if you wanted to for the user to get interface names without actually using system commands which can take a load on your cpu, use something like this

```go
package main 

/*
#include <net/if.h>
#include <stdio.h>

int main (void)
{
    struct if_nameindex *if_nidxs, *intf;

    if_nidxs = if_nameindex();
    if ( if_nidxs != NULL )
    {
        for (intf = if_nidxs; intf->if_index != 0 || intf->if_name != NULL; intf++)
        {
            printf("%s\n", intf->if_name);
        }

        if_freenameindex(if_nidxs);
    }

    return 0;
}
*/
import "C"

func main() {
     C.main()
}

```

which will grab all interface names

i would be happy to talk more and more about the things you can add and the many ways i think i could help you with this program and improving it, but i dont think i should go further at this moment in time 

if you need to contact me,  maybe ask me some question i am more active on discord which is down below 

>  Discord ID -> Scare_Sec_RE43P3R#7355

and here is a go server that can help you possibly with any issues go related 

> https://discord.gg/HNqHucTUYF

~ Kindest regards ArkAngeL43
